### DISCUSSION TITLE
Add option to automatically switch chain before sending transaction.

### DISCUSSION THREAD

**Discussion by sunwrobert** [wevm/wagmi#3867] [Ideas]
I noticed that there's a chainId parameter in sendTransaction that'll error if the current chain does not match.

```
Uncaught (in promise) TransactionExecutionError: The current chain of the wallet (id: 8453) does not match the target chain for the transaction (id: 1 – undefined).
```

Is it possible to add an option to automatically switch chain to the specified parameter if you're not currently connected to that chain?

Otherwise I'd have to create a wrapper `useSendTransaction` that manually calls `useSwitchChain` that'd satisfy my use-case when I think this feature is pretty valuable to have built in IMO.

**Comment by tmm** [comment]
Wagmi attempts to follow Unix principles and keep function atomic and composable so unlikely this gets added. Better to be explicit and call what you need when you need it.

**Reply by sunwrobert** [reply]
ah, kk, thanks for the reply either way!
